borgmatic/execute.py

```diff
@@ -213,28 +213,42 @@ def execute_command(
 
 
 def execute_command_and_capture_output(
-    full_command, capture_stderr=False, shell=False, extra_environment=None, working_directory=None,
+    full_command,
+    capture_stderr=False,
+    shell=False,
+    extra_environment=None,
+    working_directory=None,
+    raise_on_exit_code_one=True,
 ):
     '''
     Execute the given command (a sequence of command/argument strings), capturing and returning its
     output (stdout). If capture stderr is True, then capture and return stderr in addition to
     stdout. If shell is True, execute the command within a shell. If an extra environment dict is
     given, then use it to augment the current environment, and pass the result into the command. If
     a working directory is given, use that as the present working directory when running the command.
+    If raise on exit code one is False, then treat exit code 1 as a warning instead of an error.
 
-    Raise subprocesses.CalledProcessError if an error occurs while running the command.
+    Raise subprocesses.CalledProcessError if an error occurs while running the command, or if the
+    command exits with a non-zero exit code and raise on exit code one is True.
     '''
     log_command(full_command)
     environment = {**os.environ, **extra_environment} if extra_environment else None
     command = ' '.join(full_command) if shell else full_command
 
-    output = subprocess.check_output(
-        command,
-        stderr=subprocess.STDOUT if capture_stderr else None,
-        shell=shell,
-        env=environment,
-        cwd=working_directory,
-    )
+    try:
+        output = subprocess.check_output(
+            command,
+            stderr=subprocess.STDOUT if capture_stderr else None,
+            shell=shell,
+            env=environment,
+            cwd=working_directory,
+        )
+        logger.warning('Command output: {}'.format(output))
+    except subprocess.CalledProcessError as e:
+        if raise_on_exit_code_one or e.returncode != 1:
+            raise
```

Comment:
This logic looks correct to me, but it's effectively repeating some of the logic encapsulated in `exit_code_indicates_error()`, which has indeed changed over time as the concept of "what's an error" has evolved with different Borg releases. So it would be great to continue to keep that logic in one place.

To do that, maybe you could call `exit_code_indicates_error()` as needed from `execute_command_and_capture_output()`? Right now `exit_code_indicates_error()` takes a `process` as a parameter, but there's no reason it needs to. It could just as easily take a command sequence (which you have here), and then other callers could pass in `process.args` for that command.

I know this expands scope a little, but hopefully not too much as it's mostly a mechanical change.